tests/unit/storage/test_gcp.py

```diff
@@ -119,32 +173,41 @@ def test_write_then_read_file_gzipped(self, codecov_vcr):
     def test_read_file_does_not_exist(self, request, codecov_vcr):
         storage = GCPStorageService(gcp_config)
         path = f"{request.node.name}/does_not_exist.txt"
-        bucket_name = "testingarchive004"
+        bucket_name = "testingarchive1"
         with pytest.raises(FileNotInStorageError):
             storage.read_file(bucket_name, path)
 
     def test_read_file_application_gzip(self, request, codecov_vcr):
         storage = GCPStorageService(gcp_config)
         path = "gzipped_file/test_006.txt"
-        bucket_name = "testingarchive004"
+        bucket_name = "testingarchive1"
+        content_to_upload = "content to write\nThis is crazy\nWhy does this work"
+        bucket = storage.storage_client.get_bucket(bucket_name)
+        blob = google_storage.Blob(path, bucket)
+        blob.content_encoding = "gzip"
+        f = gzip.compress(content_to_upload.encode())
+        blob.upload_from_string(f, content_type="text/plain")
+        content = storage.read_file(bucket_name, path)
+        assert content.decode() == content_to_upload
+
+    def test_read_file_application_zstd(self, request, codecov_vcr):
+        storage = GCPStorageService(gcp_config)
+        path = "zstd_test.txt"
+        bucket_name = "testingarchive1"
         content_to_upload = "content to write\nThis is crazy\nWhy does this work"
         bucket = storage.storage_client.get_bucket(bucket_name)
         blob = google_storage.Blob(path, bucket)
-        with io.BytesIO() as f:
-            with gzip.GzipFile(fileobj=f, mode="wb", compresslevel=9) as fgz:
-                fgz.write(content_to_upload.encode())
-            blob.content_encoding = "gzip"
-            blob.upload_from_file(
-                f, size=f.tell(), rewind=True, content_type="application/x-gzip"
```

Comment:
I don’t think the "file" part of this test is important, but the `application/x-gzip` part I believe is. When I was messing with this code, I found out that the crc32 check was failing depending on the content_type, and I believe that is the reason there was some workaround for it specifically in the code.
We should make sure things are still working as expected there.

Reply:
this makes sense and i will take a look